gui/symfony-docker/src/Service/BlockChainService.php

```diff
@@ -126,6 +132,30 @@ public function getAllRessourceFromWalletAddress(String $WalletAddress, String $
         return $returnData;
     }
 
+    public function getRessourceFromTokenId(int $tokenId): array
+    {
+        $data = $this->getMetaDataFromTokenId($tokenId);
+        $returnData = [
+            "tokenID" => $tokenId,
+            "resourceID" => $data["resource_id"],
+            "resourceName" => $data["resource_name"],
+            "resourceType" => "ANIMAL", //need to be change later
+            // "quantity" => $data["data"][0]["balance"],
+            "isContaminated" => $data["data"][0]["stringData"]["isContaminated"],
+            "weight" => $data["data"][0]["stringData"]["weight"],
+            "price" => $data["data"][0]["stringData"]["price"],
+            "description" => $data["data"][0]["stringData"]["description"],
+            "genre" => $data["data"][0]["stringData"]["genre"],
+            "address" => $data["data"][0]["stringData"]["address"],
+            "birthDate" => $data["data"][0]["stringData"]["birthDate"],
+            "nutrition" => $data["data"][0]["stringData"]["nutrition"],
+            "vaccin" => $data["data"][0]["stringData"]["vaccin"],
+        ];
+        return $returnData;
+    }
+
+    
+
     public function getStringDataFromTokenID(int $tokenId): array
```

Comment:
what's the difference between these two functions please ? 
- getStringDataFromTokenID(int $tokenId)
- getMetaDataFromTokenId($tokenId)

Reply:
Metadata is everything in the call "http://127.0.0.1:8080/resource/".$tokenId."/metadata"
StringData is the short part of the data that i can writte and is make by me, i can change is whenever i want, don t need to ask etienne